### PR TITLE
Fix false positives on non-standard syntax in declaration-block-no-*

### PR DESCRIPTION
**Pull request by gucong3000**
> Which issue, if any, is this issue related to?

#3380 
emotion-js/emotion#686 (comment)

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.js [hunk 1]
```diff
@@ -35,6 +35,11 @@ testRule(rule, {
   ],
 
   reject: [
+    {
+      code:
+        "margin-left: 10px; margin-right: 10px; margin-top: 20px; margin-bottom: 30px;",
+      message: messages.expected("margin")
+    },
```
Comment (evilebottnawi) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.js]: Let's add test with html and style tag

Comment (gucong3000) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.js]: Done.

File: lib/rules/declaration-block-no-duplicate-properties/__tests__/index.js [hunk 2]
```diff
@@ -18,6 +18,10 @@ testRule(rule, {
       code: "a { color: pink; { &:hover { color: orange; } } }",
       description: "spec nested"
     },
+    {
+      code: "color: pink; @media { color: orange; }",
```
Comment (jeddy3) [inline lib/rules/declaration-block-no-duplicate-properties/__tests__/index.js]: As per [the developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#write-tests):

> Ensure you use standard CSS syntax by default

I don't believe a declaration outside of rule-set is valid. What are we trying to test here?

If it's the content of a `style=""` attribute then how does this differ from the `syntax: html` tests below which are more explicit?

File: lib/rules/declaration-block-no-duplicate-properties/__tests__/index.js [hunk 3]
```diff
@@ -48,6 +52,10 @@ testRule(rule, {
   ],
 
   reject: [
+    {
+      code: "color: pink; color: orange",
```
Comment (jeddy3) [inline lib/rules/declaration-block-no-duplicate-properties/__tests__/index.js]: If this is for `style=""` attributes? If so, how does it differ from the `syntax: html` tests below?

**Comment by jeddy3** [comment]
@evilebottnawi If you've time to review this today, then I'll be able to do a release before I go away for the weekend.

**Comment by evilebottnawi** [comment]
@jeddy3 LGTM

**Comment by jeddy3** [comment]
> -   Fixed: `declaration-block-no-*` false positives for non-standard syntax ([#3381](https://github.com/stylelint/stylelint/pull/3381)).

**Pull request merged**